lib/javadoc-extension.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const toProc = require('./util/to-proc')
+
+const METHOD_REGEX = /(.*)\((.*)\)/
+
+function register (registry, context) {
+  if (!(registry && context)) return // NOTE only works as scoped extension for now
+  registry.$groups().$store('springio/javadoc', toProc(createExtensionGroup(context)))
+  return registry
+}
+
+function createExtensionGroup () {
+  return function () {
+    this.inlineMacro(function () {
+      this.named('javadoc')
+      this.process((parent, target, attrs) => {
+        const inline = process(parent.getDocument(), parseTarget(target), attrs)
+        return this.createInline(parent, 'quoted', inline.text, {
+          type: 'monospaced',
+          attributes: { role: inline.role },
+        })
+      })
+    })
+  }
+}
+
+function parseTarget (target) {
+  target = target.replaceAll('&#8230;&#8203;', '...')
+  const lastSlash = target.lastIndexOf('/')
+  const location = lastSlash !== -1 ? target.substring(0, lastSlash) : undefined
+  const reference = lastSlash !== -1 ? target.substring(lastSlash + 1, target.length) : target
+  const lastHash = reference.lastIndexOf('#')
+  const classReference = lastHash !== -1 ? reference.substring(0, lastHash) : reference
+  const anchor = lastHash !== -1 ? reference.substring(lastHash + 1, reference.length) : undefined
+  return { location, classReference, anchor }
+}
+
+function process (document, target, attrs) {
+  const location = target.location || document.getAttribute('javadoc-location', 'xref:attachment$api/java')
+  const format = attrs.format || document.getAttribute('javadoc-format', 'short')
+  const linkLocation = link(location, target)
+  const listDescription = applyFormat(target, format, attrs.$positional)
```

Comment:
Very minor, but I think that this might be a typo and should be `linkDescription`